Src/CSharpier.VSCode/src/CSharpierProcessProvider.ts

```diff
@@ -67,15 +67,35 @@ export class CSharpierProcessProvider implements Disposable {
 
         if (!version || !this.csharpierProcessesByVersion[version]) {
             // this shouldn't really happen, but just in case
+            this.logger.debug(`returning NullCSharpierProcess because there was no csharpierProcessesByVersion for ${version}`);
             return new NullCSharpierProcess();
         }
 
         return this.csharpierProcessesByVersion[version]!;
     };
 
+    private getDirectoryOfFile = (filePath: string) => {
+        let directory = path.parse(filePath).dir;
+        if (directory) {
+            return directory;
+        }
+
+        if (workspace.workspaceFolders && workspace.workspaceFolders.length > 0) {
+            directory = workspace.workspaceFolders[0].uri.fsPath
+            this.logger.debug("Unsaved document has no directory, falling back to workspace folder: " + directory);
+        }
+        else {
+            directory = __dirname;
+            this.logger.debug("Unsaved document has no directory, falling back to __dirname: " + directory);
+        }
+
+        return directory;
+    };
+
     private getCSharpierVersion = (directoryThatContainsFile: string): string => {
         let currentDirectory = directoryThatContainsFile;
-        while (true) {
+        let loopNumber = 0;
```

Comment:
Okay, that's not a _better_ name :stuck_out_tongue: 
How about `parentNumber`?

Or better yet, how about stopping at the filesystem root instead of an arbitrary count?
https://stackoverflow.com/a/70115897